### PR TITLE
Refactor trapezoid construction

### PR DESCRIPTION
**Pull request by sethrj**
This is a minor follow-on to #1218 that does some refactoring for improved clarity and accuracy.

### CODE REVIEW

File: src/orange/orangeinp/ConvexRegion.cc [hunk 1]
```diff
@@ -338,56 +338,65 @@ GenTrap GenTrap::from_trd(real_type halfz, Real2 const& lo, Real2 const& hi)
 
 //---------------------------------------------------------------------------//
 /*!
- * Construct from half Z height and 1-4 vertices for top and bottom planes.
+ * Construct from skewed trapezoids.
+ *
+ * See
+ * https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/Detector/Geometry/geomSolids.html#constructed-solid-geometry-csg-solids
+ * for details on construction.
```
Comment (mrguilima) [inline src/orange/orangeinp/ConvexRegion.cc]:
```suggestion
 * For details on construction, see
 * https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/Detector/Geometry/geomSolids.html#constructed-solid-geometry-csg-solids
```

**Review by mrguilima — APPROVED** [review]
Loogs good to me.
Thanks!

File: src/orange/g4org/SolidConverter.cc [hunk 2]
```diff
@@ -105,6 +95,64 @@ SolidEnclosedAngle get_polar_wedge(S const& solid)
...
+//---------------------------------------------------------------------------//
+/*!
+ * Return theta, phi angles for a G4Para or G4Trap given their symmetry axis.
+ *
+ * Certain Geant4 shapes are constructed by skewing the z axis and providing
+ * the polar/azimuthal angle of the transformed axis. This calculates that
+ * transform by converting from cartesian to spherical coordinates.
+ *
+ * The components of the symmetry axis for G4Para/Trap are always encoded as a
+ * vector \f$ (A \tan(\theta)\cos(\phi), A \tan(\theta)\sin(phi), A) \f$.
```
Comment (mrguilima) [inline src/orange/g4org/SolidConverter.cc]:
```suggestion
 * vector \f$ (A \tan(\theta)\cos(\phi), A \tan(\theta)\sin(\phi), A) \f$.
```

**Comment by sethrj** [comment]
Thanks for the speedy review and feedback @mrguilima !

**Pull request merged**